utilities/tools/module-indexes/module-index.tests.ps1

```diff
@@ -0,0 +1,747 @@
+                }
+                $lineNumber++
+            }
+        }
+
+        It 'Should not have any leading white spaces in any value' {
+            $lineNumber = 1
+            foreach ($line in $rawFile) {
+                $line | Should -Not -Match '^\s+|,\s+' -Because "there should not be any leading white spaces anywhere in line #$lineNumber"
+                $lineNumber++
+            }
+        }
+
+
+    }
+
+    Context 'ModuleDisplayName column' {
+
+        It "Should not have any missing values in the 'ModuleDisplayName' column" {
+            $lineNumber = 2
+            foreach ($item in $csvContent) {
+                $item.ModuleDisplayName | Should -Not -BeNullOrEmpty -Because "ModuleDisplayName is a required field. In line #$lineNumber, this is empty. The following values have been provided: ""$($item)"""
+                $lineNumber++
+            }
+        }
+
+        It "Should not have any duplicate values in the 'ModuleDisplayName' column" {
+            $duplicates = $csvContent | Group-Object -Property ModuleDisplayName | Where-Object Count -GT 1
+            $duplicates | Should -BeNullOrEmpty -Because "each module's ModuleDisplayName should be unique. This is a duplicate: ""$($duplicates.Name)"""
+        }
+    }
+
+    Context 'ModuleName column' {
+        It "Should not have any missing values in the 'ModuleName' column" {
+            $lineNumber = 2
+            foreach ($item in $csvContent) {
+                $item.ModuleName | Should -Not -BeNullOrEmpty -Because "ModuleName is a required field. In line #$lineNumber, this is empty. The following values have been provided: ""$($item)"""
+                $lineNumber++
+            }
+        }
+
+        It "Should not have any duplicate values in the 'ModuleName' column" {
+            $duplicates = $csvContent | Group-Object -Property ModuleName | Where-Object Count -GT 1
+            $duplicates | Should -BeNullOrEmpty -Because "each module's ModuleName should be unique. This is a duplicate: ""$($duplicates.Name)"""
+        }
+
+        if ($CsvFilePath -match 'Bicep') {
+
+            It "Should only contain allowed characters" {
+                $lineNumber = 2
+                foreach ($item in $csvContent) {
+                    $item.ModuleName | Should -Match '^[a-z0-9]+(?:[\/-][a-z0-9]+)*$' -Because "each segment of ModuleName should only contain lowercase letters, numbers and hyphens, in line #$lineNumber"
+                    $lineNumber++
+                }
+            }
+
+            if ($CsvFilePath -match 'ResourceModules') {
+                It "Should start with 'avm/res/'" {
+                    $lineNumber = 2
+                    foreach ($item in $csvContent) {
+                        $item.ModuleName | Should -Match '^avm/res/.*' -Because "ModuleName should start with 'avm/res/'. In line #$lineNumber, this is invalid. The following values have been provided: ""$($item.ModuleName)"""
```

Comment:
```suggestion
                        $item.ModuleName | Should -Match '^avm\/res\/.*' -Because "ModuleName should start with 'avm/res/'. In line #$lineNumber, this is invalid. The following values have been provided: ""$($item.ModuleName)"""
```

Reply:
Suggestion applied.